File: Data Structures & Algorithms/is-anagram/submission-0.cpp
#include <map>

class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char, int> m1{ stringToMap(s) };
        map<char, int> m2{ stringToMap(t) };
        return m1 == m2;
    }

    std::map<char, int> stringToMap(string s) {
        map<char, int> m;
        for (char c: s) {
            if (m.find(c) != m.end()) { m[c] += 1; }
            else { m[c] = 1; }
        } 
        return m;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-0.py
class Solution:
    def trap(self, height: List[int]) -> int:
        # find greatest values to left and right of 0, minimum of both
        # subtract by amount of ground

        n = len(height)
        
        l = [0 for i in range(n)] 
        m = -1
        for i in range(1,n-1):
            m = max(m, height[i-1])
            l[i]=m

        r = [0 for i in range(n)] 
        m = -1
        for j in range(n-2,0,-1):
            m = max(m, height[j+1])
            r[j] = m

        area = 0
        for k in range(n):
            ith = min(l[k],r[k])-height[k]
            if ith > 0:
                area += ith
        
        return area
